src/compiler/rust.rs

```diff
@@ -34,15 +46,29 @@ use std::io;
 use std::io::Read;
 use std::iter;
 use std::path::{Path, PathBuf};
-use std::process::Stdio;
-use std::time::Instant;
+use std::process;
+#[cfg(feature = "dist-client")]
+use std::sync::{Arc, Mutex};
+use std::time;
 use tempdir::TempDir;
 use util::{fmt_duration_as_secs, run_input_output, Digest};
 use util::{HashToDigest, OsStrExt, ref_env};
 
 use errors::*;
 
+/// Can dylibs (like proc macros) be distributed on this platform?
+#[cfg(all(feature = "dist-client", target_os = "linux", target_arch = "x86_64"))]
+const CAN_DIST_DYLIBS: bool = true;
+#[cfg(all(feature = "dist-client", not(all(target_os = "linux", target_arch = "x86_64"))))]
+const CAN_DIST_DYLIBS: bool = false;
+
+#[cfg(feature = "dist-client")]
+const RLIB_PREFIX: &str = "lib";
+#[cfg(feature = "dist-client")]
```

Comment:
consts like these would probably be fine with just `#[allow(unused)]`, but it's not a big deal.